File: api/csharp/examples/MQ/Consumer.cs
using Zbus.Net;
using Zbus.Mq;
using Zbus.Broker;

namespace Zbus.Examples.MQ
{

    class MyHandler : IMessageHandler
    {
        public void Handle(Message msg, Consumer consumer)
        {
            System.Console.WriteLine(msg);
        }
    }

    class ConsumerTest
    {
        public static void Main(string[] args)
        {  
            IBroker broker = new SingleBroker(); //using BrokerConfig to change default

            Consumer c = new Consumer(broker, "MyMQ");
            c.ConsumeTimeout = 30000;

            c.OnMessage(new MyHandler());
            c.Start();
            //c.Stop();

            //broker.Dispose();
        }
    }
}

File: clients/Javascript/examples/Consumer.js
if (typeof module !== 'undefined' && module.exports) {
    var zbus = require("../../../src/main/resources/zbus.js"); 
    var Broker = zbus.Broker;
    var Consumer = zbus.Consumer;
}

var broker = new Broker();
broker.addTracker("localhost:15555");

var c = new Consumer(broker, "hong6"); 
c.onMessage = function (msg, client) {
	console.log(msg);
}
c.start();  
